### PR TITLE
Create multiple tasks when uploading multiple videos

### LINKED ISSUE #916: Create a lot of tasks with videos at the same time
I have about 240 videos which I want to tag. 

But to create a single task for each video is very time-consuming. I would like to upload several videos and edit them all together.

### PR DESCRIPTION
**Pull request by AlexeyAlexeevXperienceAI**
### Motivation and context

Resolve #916

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by bsekachev** [comment]
Hi, Here are some implementation issues/ideas I came across:

- That is not really obvious how to see these template rules.
I would suggest adding an QuestionCircle icon with this tooltip, or maybe somehow highlight "template" word (making it a hyperlink style?)
image

- Let's add spaces here. BTW, how do ``pending`` work? It is always 0 for me, even if I create several tasks with 1 rq worker.
image

- I am not sure we need this pop-up notification, looks useless because we have a progress bar near the submit button:
image

- Can we somehow escape ``Database is locked`` on development environment?

- I was able to create a task with .png file (but if I select several .png files, I can't):
image

- When created tasks from connected file share, I have extra slash at the begging of a task name:
image

- Where ``Cloud storage`` option is? 
image

- With remote sources I also can submit tasks with .jpeg files:
image

- Strange scenario:
1. Put two links (to images) to Remote sources
2. It shows the error (Only bulk of videos)
3. Remove one of links
4. The error disappears
5. Press "Submit 1 task" button
6. It shows progress bar with 2 tasks
7. One of tasks creation fails and UI is fails
image
8. The second task with an image is successfully created

**Comment by bsekachev** [comment]
Could you please provide any source & license for ``multi-plus-icon.svg``?
Usually if we get icons outside, we put source & license to the beginning of .svg file.

File: cvat-ui/src/actions/tasks-actions.ts [hunk 1]
```diff
@@ -402,26 +357,18 @@ export function createTaskAsync(data: any): ThunkAction<Promise<void>, {}, {}, A
...
                 gitPlugin.data.repos = data.advanced.repository;
                 gitPlugin.data.format = data.advanced.format;
                 gitPlugin.data.lfs = data.advanced.lfs;
             }
         }
-
-        dispatch(createTask());
-        try {
-            const savedTask = await taskInstance.save((status: string, progress: number): void => {
-                dispatch(createTaskUpdateStatus(status + (progress !== null ? ` ${Math.floor(progress * 100)}%` : '')));
```
Comment (bsekachev) [inline cvat-ui/src/actions/tasks-actions.ts:417]: We should not lost progress value, it is useful to a user

File: cvat-ui/src/components/create-task-page/basic-configuration-form.tsx [hunk 2]
```diff
@@ -4,15 +4,19 @@
...
-import { Store } from 'antd/lib/form/interface';
 
 export interface BaseConfiguration {
     name: string;
 }
 
 interface Props {
-    onSubmit(values: BaseConfiguration): void;
+    onChange(values: BaseConfiguration): void;
+    isMultiTask: boolean;
```
Comment (bsekachev) [inline cvat-ui/src/components/create-task-page/basic-configuration-form.tsx:17]: Let's have ``many: boolean;``

File: cvat-ui/src/components/create-task-page/basic-configuration-form.tsx [hunk 3]
```diff
@@ -4,15 +4,19 @@
...
+    exampleMultiTaskName?: string;
+    defaultValue?: string;
```
Comment (bsekachev) [inline cvat-ui/src/components/create-task-page/basic-configuration-form.tsx:19]: I see the following logic: 
`` const defaultValue = isMultiTask ? '{{file_name}}' : '';``

isMultiTask is also a prop of this component, so, let's put this logic inside and remove this extra prop.

**Comment by AlexeyAlexeevXperienceAI** [comment]
@bsekachev 
> Let's add spaces here. BTW, how do pending work? It is always 0 for me, even if I create several tasks with 1 rq worker.
not reproduced, please check if you still have this issue 
> Can we somehow escape Database is locked on development environment?
decided to make one queue for stability
> Where Cloud storage option is?
Cloud storage doesn’t work with video and hides in multi-mode

Other issues is fixed

**Comment by bsekachev** [comment]
Is the PR ready to review?

**Comment by bsekachev** [comment]
> Cloud storage doesn’t work with video and hides in multi-mode

Could you please add a couple of lines to the PR description about the restrictions? 

> not reproduced, please check if you still have this issue

Not reproduced anymore.

Why total is 5?
image

One of tasks is failed:
image

Looks like it considers a directory like a task:
image

**Comment by bsekachev** [comment]
If I select a directory with videos, it does not work.
I understand why, UI does not now about any files inside. Maybe we should strict checking directories in multiple tasks creation mode?

image

**Comment by kirill-sizov** [comment]
@bsekachev I closed full check, because e2e tests was executing extremely long. I need some time to investigate it.

**Comment by bsekachev** [comment]
@kirill-sizov 

Why do you think so, current run took about 25 minutes, yesterday as I noticed GitHub had service issues.

**Comment by bsekachev** [comment]
UX issues I currently see:
- In the shared storage we are able to check directories, it leads to errors (see my comment above)
- 'Cancel' button does not work for a task which is ``in progress``

**Pull request merged**